packages/block-library/src/paragraph/edit.js

```diff
@@ -55,8 +61,25 @@ function ParagraphBlock( {
 } ) {
 	const { align, content, direction, dropCap, placeholder } = attributes;
 	const isDropCapFeatureEnabled = useSetting( 'typography.dropCap' );
+	const [ paragraphElement, setParagraphElement ] = useState( null );
+	const [ isDragging, setIsDragging ] = useState( false );
+	const refCallback = ( element ) => {
+		setParagraphElement( element );
+	};
+	const draggingRef = useDropZone( {
+		onDragStart: () => {
+			setIsDragging( true );
+		},
+		onDragEnd: () => {
+			setIsDragging( false );
+		},
+	} );
 	const blockProps = useBlockProps( {
-		ref: useOnEnter( { clientId, content } ),
+		ref: useMergeRefs( [
+			useOnEnter( { clientId, content } ),
+			refCallback,
```

Comment:
React has no guarantee that the signature of `setState` will always match the signature of the ref callback. Though I also doubt that it would ever change 😅. I'm fine either way.